src/designspace.rs

```diff
@@ -208,6 +210,17 @@ mod serde_impls {
         }
         Helper::deserialize(deserializer).map(|x| x.source)
     }
+
+    /// Generate a unique name for a source.
+    ///
+    /// We do not make guarantees about what name will be assigned to a given
+    /// source, only that it will be unique within this process. If the designer
+    /// cares, they should provide a name explicitly.
+    pub fn generate_missing_source_name() -> String {
+        static COUNTER: AtomicU64 = AtomicU64::new(0);
+        let next_n = COUNTER.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
+        format!("unnamed_source_{next_n}")
```

Comment:
Optional, we could match the existing authoritative implementation (https://github.com/fonttools/fonttools/blob/main/Lib/fontTools/designspaceLib/__init__.py#L2311), though I cannot think of a reason it would matter.

Reply:
this would be more annoying, since we don't have access to serde's internal state (and hence don't know the actual index of the current source) and so we would need to do a post-processing step to assign the names after loading the whole designspace. 🤷